wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/server/AuthorizationSupplier.java

```diff
@@ -14,8 +14,11 @@
 import org.eclipse.che.api.factory.server.scm.exception.ScmCommunicationException;
 import org.eclipse.che.api.factory.server.scm.exception.ScmUnauthorizedException;
 
-/** Compute the Authorization header to sign the OAuth 1 request. */
-public interface AuthorizationHeaderSupplier {
+public interface AuthorizationSupplier {
+  /** Compute the Authorization header to sign the OAuth 1 request. */
   String computeAuthorizationHeader(final String requestMethod, final String requestUrl)
       throws ScmUnauthorizedException, ScmCommunicationException;
+
+  /** Construct ScmUnauthorizedException linked to authorization method. */
+  ScmUnauthorizedException buildScmUnauthorizedException();
```

Comment:
this method does not feel right to be in the interface. looks more like helper method.